src/libs/ajax/User.ts

```diff
@@ -192,17 +196,50 @@ export const User = (signal?: AbortSignal) => {
       return res.json();
     },
 
-    getEnterpriseFeatures: async (): Promise<string[]> => {
-      try {
-        const res = await fetchSam(
-          '/api/resources/v2?format=flat&resourceTypes=enterprise-feature&roles=user',
-          _.mergeAll([authOpts(), { signal }])
-        );
-        const json = await res.json();
-        return json.resources.map((resource) => resource.resourceId);
-      } catch (error: unknown) {
-        return [];
-      }
+    getSamUserCombinedState: async (): Promise<SamUserCombinedStateResponse> => {
+      const res = await fetchSam('api/users/v2/self/combinedState', _.mergeAll([authOpts(), { signal }]));
+      const responseJson = await res.json();
+      const samUser = {
+        id: responseJson.samUser.id,
+        googleSubjectId: responseJson.samUser.googleSubjectId,
+        email: responseJson.samUser.email,
+        azureB2CId: responseJson.samUser.azureB2CId,
+        allowed: responseJson.samUser.allowed,
+        createdAt: responseJson.samUser.createdAt ? new Date(responseJson.samUser.createdAt) : undefined,
+        registeredAt: responseJson.samUser.registeredAt ? new Date(responseJson.samUser.registeredAt) : undefined,
+        updatedAt: responseJson.samUser.updatedAt ? new Date(responseJson.samUser.updatedAt) : undefined,
+      } as SamUserResponse;
+
+      const terraUserAllowances = {
+        allowed: responseJson.allowances.allowed,
+        details: {
+          enabled: responseJson.allowances.details.enabled,
+          termsOfService: responseJson.allowances.details.termsOfService,
+        } as SamUserAllowancesDetails,
+      } as SamUserAllowances;
+
+      const terraUserAttributes = { marketingConsent: responseJson.attributes.marketingConsent } as SamUserAttributes;
+
+      const termsOfService = {
+        latestAcceptedVersion: responseJson.termsOfServiceDetails.latestAcceptedVersion,
+        acceptedOn: responseJson.termsOfServiceDetails.acceptedOn
+          ? new Date(responseJson.termsOfServiceDetails.acceptedOn)
+          : undefined,
+        permitsSystemUsage: responseJson.termsOfServiceDetails.permitsSystemUsage,
+        isCurrentVersion: responseJson.termsOfServiceDetails.isCurrentVersion,
+      } as SamUserTermsOfServiceDetails;
+
+      const enterpriseFeatures = responseJson.additionalDetails.enterpriseFeatures
+        ? responseJson.additionalDetails.enterpriseFeatures.resources.map((resource) => resource.resourceId)
+        : [];
+
+      return {
+        samUser,
+        terraUserAllowances,
+        terraUserAttributes,
+        termsOfService,
+        enterpriseFeatures,
+      } as SamUserCombinedStateResponse;
```

Comment:
No cast necessary here since the object does conform to the type.

```suggestion
      };
```